pyresample/future/geometry/area.py

```diff
@@ -19,4 +19,87 @@
 
 from __future__ import annotations
 
-from pyresample.geometry import AreaDefinition, DynamicAreaDefinition  # noqa
+from typing import Optional, Union
+
+from pyproj import CRS
+
+from pyresample.geometry import AreaDefinition as LegacyAreaDefinition  # noqa
+from pyresample.geometry import (  # noqa
+    DynamicAreaDefinition,
+    get_geostationary_angle_extent,
+    get_geostationary_bounding_box,
+)
+
+
+class AreaDefinition(LegacyAreaDefinition):
```

Comment:
Do we want the new area def to inherit all the methods of the legacy one, or do we want to split responsibility a bit?

Reply:
Yeah I struggled with this decision. Initially this class in the `future` subpackage didn't have anything custom about it so it was just a basic import. I think for ease of transitioning to future we might want it to be a subclass so that `isinstance` checks still work? Or maybe usages of stuff like that should be fixed to be more duck-typing-friendly?

Reply:
I vote for more duck typing.

Reply:
Please read my other comments. That is the plan, but I don't want to throw that all into one PR and I also don't want to deal with all the tests failing because I didn't implement one of 50 methods on the legacy AreaDefinition class. The plan is to do this work in one of the next 2 or 3 PRs.